code/game/objects/explosion.dm

```diff
@@ -95,6 +95,27 @@
 		for (var/turf/T in trange(max_range, epicenter))
 			var/dist = cheap_pythag(T.x - x0, T.y - y0)
 
+			if(explosion_newmethod)	//realistic explosions that take obstacles into account
+				var/turf/Trajectory = T

```

Comment:
Yes, that's good, but i you look where this definition is, you're defining a turf as a turf.
```var/turf/Trajectory = var/turf/T```